### PR TITLE
feat: move util functions from web

### PR DESCRIPTION
**Pull request by elmutt**
Move utils functions from web to lib

### CODE REVIEW

File: packages/chain-adapters/src/bitcoin/BitcoinChainAdapter.ts [hunk 1]
```diff
@@ -197,7 +197,7 @@ export class ChainAdapter implements IChainAdapter<ChainTypes.Bitcoin> {
             addressNList: bip32ToAddressNList(
               `${path}/1/${String(account.chainSpecific.nextChangeAddressIndex)}`
             ),
-            scriptType: BTCOutputScriptType.PayToWitness,
+            scriptType: scriptType as unknown as BTCOutputScriptType,
```
Comment (reidrankin) [inline packages/chain-adapters/src/bitcoin/BitcoinChainAdapter.ts:200]: this seems dangerous; what's going on here?

File: packages/chain-adapters/src/utils/utxoUtils.ts [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    }
+  }
+}
+
+/**
+ * Returns coin purpose from a given script type
+ * @param scriptType
+ * @returns 44 | 49 | 84
+ */
+export const purposeFromScript = (
```
Comment (reidrankin) [inline packages/chain-adapters/src/utils/utxoUtils.ts:34]: a purpose should not be derived from a script type; to do so assumes not only that every wallet follows BIP44, but that another BIP will always allocate a fresh purpose number to go with a new script type, and that's not guaranteed.

File: packages/chain-adapters/src/utils/utxoUtils.ts [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+}
+
+/**
+ * Utility function to get params to get BIP32Params and scriptType for chain-adapter functions (getAddress, buildSendTransaction)
+ * Get return an object with utxo bip32Params derived from script type and asset
+ * @param scriptType
+ * @param asset
+ * @returns object with BIP32Params and scriptType or undefined
+ */
+export const bip32AndScript = (
```
Comment (reidrankin) [inline packages/chain-adapters/src/utils/utxoUtils.ts:55]: I'm not sure what problem you're trying to solve here, but I don't think it should be solved this way.
- there's no need to return the script type if the caller knows it already
- you shouldn't get bip32params from an asset; you might be able to get a _default_ bip32 path from an asset, but an asset may be stored at any other path, and in the case of change addresses _definitely is_ in the normal use case.

**Review by reidrankin — DISMISSED** [review]

File: packages/chain-adapters/src/utils/utxoUtils.ts (hunk 2 above)
Comment (elmutt) [reply packages/chain-adapters/src/utils/utxoUtils.ts:34]: what if I call it "bip44 purpose from script"?

File: packages/chain-adapters/src/utils/utxoUtils.ts (hunk 3 above)
Comment (elmutt) [reply packages/chain-adapters/src/utils/utxoUtils.ts:55]: I return the script thype so I can easily spread it and bip32Params into chain-adapter calls  in on easy line where its needed. namely getAddress and buildSendTx.

**Review by cjthompson — APPROVED** [review]

File: packages/types/src/base.ts [hunk 4]
```diff
@@ -31,6 +31,12 @@ export enum NetworkTypes {
   ETH_RINKEBY = 'ETH_RINKEBY'
 }
 
+export enum UtxoAccountType {
+  SegwitNative = 'SegwitNative',
+  SegwitP2sh = 'SegwitP2sh',
+  P2pkh = 'P2pkh'
+}
```
Comment (reidrankin) [inline packages/types/src/base.ts:38]: I'm still fundamentally unhappy about the existence (and capitalization!) of this enum, but I can't think of anything better and l'll get over it.

**Review by reidrankin — APPROVED** [review]

**Pull request merged**